src/Http/Http.Features/src/IHttpRequestTrailersFeature.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Http.Features
+{
+    /// <summary>
+    /// This feature exposes HTTP request trailer headers, either for HTTP/1.1 chunked bodies or HTTP/2 trailing headers.
+    /// </summary>
+    public interface IHttpRequestTrailersFeature
+    {
+        /// <summary>
+        /// Indicates if the <see cref="Trailers"/> are available yet. They may not be available until the
+        /// request body is fully read.
+        /// </summary>
+        bool Available { get; }
```

Comment:
Would this be better as a Task? It could be lazy initialized.

Reply:
A Task isn't a good fit, just waiting for it will rarely be enough, you'll deadlock yourself. If it's false then you need to take action (read the request body) before it will become true.

Reply:
I now remember this discussion. It's been too long.